src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/Login/FeatureExtensions.cs

```diff
@@ -0,0 +1,513 @@
+                        case SqlAuthenticationMethod.ActiveDirectoryPassword:
+                            workflow = TdsEnums.MSALWORKFLOW_ACTIVEDIRECTORYPASSWORD;
+                            break;
+                        case SqlAuthenticationMethod.ActiveDirectoryIntegrated:
+                            workflow = TdsEnums.MSALWORKFLOW_ACTIVEDIRECTORYINTEGRATED;
+                            break;
+                        case SqlAuthenticationMethod.ActiveDirectoryInteractive:
+                            workflow = TdsEnums.MSALWORKFLOW_ACTIVEDIRECTORYINTERACTIVE;
+                            break;
+                        case SqlAuthenticationMethod.ActiveDirectoryServicePrincipal:
+                            workflow = TdsEnums.MSALWORKFLOW_ACTIVEDIRECTORYSERVICEPRINCIPAL;
+                            break;
+                        case SqlAuthenticationMethod.ActiveDirectoryDeviceCodeFlow:
+                            workflow = TdsEnums.MSALWORKFLOW_ACTIVEDIRECTORYDEVICECODEFLOW;
+                            break;
+                        case SqlAuthenticationMethod.ActiveDirectoryManagedIdentity:
+                        case SqlAuthenticationMethod.ActiveDirectoryMSI:
+                            workflow = TdsEnums.MSALWORKFLOW_ACTIVEDIRECTORYMANAGEDIDENTITY;
+                            break;
+                        case SqlAuthenticationMethod.ActiveDirectoryDefault:
+                            workflow = TdsEnums.MSALWORKFLOW_ACTIVEDIRECTORYDEFAULT;
+                            break;
+                        case SqlAuthenticationMethod.ActiveDirectoryWorkloadIdentity:
+                            workflow = TdsEnums.MSALWORKFLOW_ACTIVEDIRECTORYWORKLOADIDENTITY;
+                            break;
+                        default:
+                            if (context.AccessTokenCallback != null)
+                            {
+                                workflow = TdsEnums.MSALWORKFLOW_ACTIVEDIRECTORYTOKENCREDENTIAL;
+                            }
+                            else
+                            {
+                                Debug.Assert(false, "Unrecognized Authentication type for fedauth MSAL request");
+                            }
+                            break;
+                    }
+
+                    await stream.WriteByteAsync(workflow, isAsync, ct);
+                    break;
+                case TdsEnums.FedAuthLibrary.SecurityToken:
+                    await stream.TdsWriter.WriteIntAsync(fedAuthFeatureData.accessToken.Length, isAsync, ct).ConfigureAwait(false);
+                    ct.ThrowIfCancellationRequested();
+                    if (isAsync)
+                    {
+                        await stream.WriteAsync(fedAuthFeatureData.accessToken.AsMemory(), ct).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        stream.Write(fedAuthFeatureData.accessToken);
+                    }
```

Comment:
Why not use only TdsWriter to write?
You need not reference stream and perform this logic here, TdsWriter already does that.

Reply:
Because I was sitting on this PR before the TdsWriter came along. I can use TdsWriter.

Reply:
I had migrated a bunch of calls to TdsWriter, this one was clearly left out